variables.tf

```diff
@@ -36,7 +36,7 @@ variable "domain_name" {
 
 variable "subject_alternative_names" {
   description = "A list of domains that should be SANs in the issued certificate"
-  type        = list(string)
+  type        = any # list(string | map(string))
```

Comment:
I have a belief that this change won't work with terragrunt because of type `any`. There is a _similar_ issue in [Route53 module](https://github.com/terraform-aws-modules/terraform-aws-route53/issues/59).

Could you change the type from `list(string)` to `list(map(string))` to prevent the issue?

PS: I have not executed this code yet.

Reply:
if we change from `any` to `list(map(string))` it will no-longer be a backwards compatible change for pure-terraform as they currently provide a `list(string)`.

if we keep the `any` and follow the [terragrunt workaround](https://github.com/terraform-aws-modules/terraform-aws-route53/blob/master/modules/records/variables.tf#L25-L35) it will be backwards compatible for pure-terraform but not for terragrunt.

if backwards compatibility is required for both pure-terraform & terragrunt the vars interface will need to be changed (something like a 2nd optional list of zone_ids but this isnt clean/reliable) or we bump major version to use `list(map(string))`.

if we going to bump the major version to use `list(map(string))`, do we want to consider any other changes at the same time?


current user input:
```hcl
subject_alternative_names = [
  "subdomain.mydomain.com",
  "www.subdomain.mydomain.com",
]
```

user input if `list(map(string))` is required but for the same result as above:
```hcl
subject_alternative_names = [
  {name = "subdomain.mydomain.com"},
  {name = "www.subdomain.mydomain.com"},
]
```

Reply:
hi @antonbabenko , have you had a chance to go over my previous reply & which direction we want to head?

Reply:
Sorry that it takes more time to process these PRs than one would expect.

I looked into #108 a few days ago, and I think the feature that this PR (#100) adds should be considered and implemented after it.

The main reason to order PRs like this is that users may want to create records using different instances of the providers (zone_id + provider). It will be easier to implement that way and update examples to show a variety of combinations.